code/src/components/app-table/AppTable.tsx

```diff
@@ -0,0 +1,35 @@
+import { ReactNode } from "react";
+
+import {
+  AppTableBody,
+  AppTableBox,
+  AppTableContainer,
+  AppTableHead,
+  AppTableRow
+} from "@/components/app-table/components";
+
+type AppTableProps<T> = {
+  bodyItems: T[];
+  headItems: string[];
+  renderBodyItem: (item: T) => ReactNode;
+  renderHeadItem: (item: string) => ReactNode;
+};
+const AppTable = <T,>({
+  bodyItems,
+  headItems,
+  renderBodyItem,
+  renderHeadItem
+}: AppTableProps<T>) => {
+  return (
+    <AppTableContainer>
+      <AppTableBox>
+        <AppTableHead>
+          <AppTableRow>{headItems.map(renderHeadItem)}</AppTableRow>
+        </AppTableHead>
+        <AppTableBody>{bodyItems.map(renderBodyItem)}</AppTableBody>
```

Comment:
I suggest rendering `<AppTableRow />` here and avoid using it outside, because someone can forget to wrap table with it

Reply:
fixed